### PR TITLE
add dao

### PR DESCRIPTION
**Pull request by acerlife**

### CODE REVIEW

File: src/main/java/com/makhnov/springbootcalculator/dao/CalculatorDao.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import javax.sql.DataSource;
+import java.sql.Connection;
+
+import static org.jooq.impl.DSL.field;
+import static org.jooq.impl.DSL.name;
+import static org.jooq.impl.DSL.table;
+
+@Repository
+public class CalculatorDao {
+    @Qualifier("dataSource")
```
Comment (VolodymyrMudryk) [inline src/main/java/com/makhnov/springbootcalculator/dao/CalculatorDao.java]: why do you need this qualifier?

File: src/main/java/com/makhnov/springbootcalculator/dao/CalculatorDao.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    @Autowired
+    private DataSource dataSource;
+
+    public void saveResult(String string) throws Exception {
+        PostfixResult postfixResult = new PostfixResult();
+        Connection connection = dataSource.getConnection();
+        DSLContext dslContext = DSL.using(connection, SQLDialect.MYSQL);
```
Comment (VolodymyrMudryk) [inline src/main/java/com/makhnov/springbootcalculator/dao/CalculatorDao.java]: it's wrong
we already discussed about it
1) create data a new data source https://stackoverflow.com/questions/28821521/configure-datasource-programmatically-in-spring-boot
2) create DSLContext as a new bean

File: src/main/java/com/makhnov/springbootcalculator/dao/CalculatorDao.java (hunk 1 above)
Comment (acerlife) [reply src/main/java/com/makhnov/springbootcalculator/dao/CalculatorDao.java]: now i don't need this

File: src/main/java/com/makhnov/springbootcalculator/dao/CalculatorDao.java (hunk 2 above)
Comment (acerlife) [reply src/main/java/com/makhnov/springbootcalculator/dao/CalculatorDao.java]: is done

File: src/main/java/com/makhnov/springbootcalculator/dao/CalculatorDao.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+import static org.jooq.impl.DSL.field;
+import static org.jooq.impl.DSL.name;
+import static org.jooq.impl.DSL.table;
+
+@Repository
+public class CalculatorDao {
+
+    @Autowired
+    private DSLContext dslContext;
```
Comment (VolodymyrMudryk) [inline src/main/java/com/makhnov/springbootcalculator/dao/CalculatorDao.java]: please use CTOR injection

File: src/main/java/com/makhnov/springbootcalculator/dao/CalculatorDao.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+    public void saveResult(String postfix_expression) throws Exception {
+        PostfixResult postfixResult = new PostfixResult();
+
+        dslContext.transaction(configuration -> {
```
Comment (VolodymyrMudryk) [inline src/main/java/com/makhnov/springbootcalculator/dao/CalculatorDao.java]: no need transaction here
you should have transaction on service layer

**Pull request closed without merging**